lib/blackout_updater.rb

```diff
@@ -0,0 +1,37 @@
+class BlackoutUpdater
+  # Service Object to update blackouts in the blackouts controller
+  def initialize(blackout:, update_p:)
+    @blackout = blackout
+    @update_p = update_p
+  end
+
+  def update!
+    return { result: nil, error: error } if error
+    update_handler
+  end
+
+  private
+
+  attr_reader :blackout, :update_p
+
+  def error
+    conflicts = Reservation.overlaps_with_date_range(update_p[:start_date],
```

Comment:
I would pull this query out into a method:

```
def conflicts
  @conflicts ||= Reservation.overlaps_with_date_range(...)
end
```

and then access the results by just calling `conflicts`. This will prevent the query from running more than once but keep the results accessible from anywhere in the class!